### PR TITLE
Chore: Login via the API by default in Cypress

### LINKED ISSUE #40451: e2e tests: use faster login-step
currently every e2e-tests logins by doing all the login-steps, we could speed these up by using a faster approach: either an API call, or logging in once and then keeping the cookie for next steps etc.

NOTE: we should verify that the login-steps work, it's just that we do not have to execute them in every e2e-test.

### PR DESCRIPTION
**Pull request by ashharrison90**
**What this PR does / why we need it**:

- adds a new optional `loginViaApi` argument to the login flow
- defaults this to true
- when true, login via an api call instead. this should be slightly faster 🤞 
- when false, use the previous behaviour of logging in via the UI
- make the smoke test login using the UI so we still have a test that does this
  - we might want to make an explicit test for this in the future? 🤔 
- NOTE: there is a small possibility this change will cause your e2e test to fail. if you were using `cy.request` immediately with a relative path, as in https://github.com/grafana/grafana/pull/43167/files#diff-5bc9f6a66b4b30398f655cedf4766eef2c31bd2fcd19e0a2f2014ffc328f9117L24, this might not work anymore if not logging in via the UI. you can either
  - switch back to logging in via the UI, using `loginViaApi = false`
  - use an absolute path in your request by prepending `${e2e.env('BASE_URL')}`

| before | after |
| --- | --- |
| image | image |

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana/issues/40451

**Special notes for your reviewer**:

### CODE REVIEW

**Review by axelavargas — APPROVED** [review]
This looks promising :), let's see how much time can we reduce on the build 🚀

**Review by joshhunt — APPROVED** [review]

File: packages/grafana-e2e/src/flows/login.ts [hunk 1]
```diff
@@ -3,20 +3,31 @@ import { e2e } from '../index';
...
+    e2e.pages.Login.submit().click();
 
-  // Local tests will have insecure credentials
-  if (password === DEFAULT_PASSWORD) {
-    e2e.pages.Login.skip().should('be.visible').click();
-  }
+    // Local tests will have insecure credentials
+    if (password === DEFAULT_PASSWORD) {
+      e2e.pages.Login.skip().should('be.visible').click();
+    }
```
Comment (joshhunt) [inline packages/grafana-e2e/src/flows/login.ts:28]: super minor nit, but i think code is easier to read when there's less nesting. 

Perhaps something like

```ts
if (!loginViaApi) {
    e2e().logToConsole('Logging in with username:', username);
    e2e.pages.Login.visit();
    e2e.pages.Login.username()
    etc
    return;
}

 cy.request({
      method: 'POST',
      url: `${e2e.env('BASE_URL')}/login`,
      body: {
        user: username,
        password,
      },
    });
```

or even like 

```ts
if (loginViaApi) {
  apiLoginStrategy()
} else {
  manualLoginStrategy()
```

dunno.

**Comment by ashharrison90** [comment]
for some reason the exemplars test doesn't have a `BASE_URL` set.  anyone in @grafana/user-essentials know what's going on there? 😕

**Comment by ashharrison90** [comment]
fixed it! for reference, it's because it was assuming the browser will be open at that point (because the login step would open a browser), but with api login it isn't. so we have to explicitly add the base url to that request.

**Review by hugohaggmark — APPROVED** [review]
Great work on this 🎉

File: packages/grafana-e2e/src/flows/login.ts (hunk 1 above)
Comment (hugohaggmark) [reply packages/grafana-e2e/src/flows/login.ts]: I agree with @joshhunt on this, especially breaking the code up into 2 separate functions 👍

**Pull request merged**